client/components/course/Sidebar/Relationship.vue

```diff
@@ -73,9 +92,10 @@ export default {
     getAssociationIds(activity) {
       return get(activity, `refs.${this.type}`, []);
     },
-    onRelationshipChanged(associations) {
+    onRelationshipChanged(value) {
+      const associations = Array.isArray(value) ? value : [value];
       let activity = cloneDeep(this.activity) || {};
-      set(activity, `refs.${this.type}`, map(associations, 'id'));
+      set(activity, `refs.${this.type}`, filter(map(associations, 'id')));
```

Comment:
What are you filtering here? 🙄

Reply:
When deselecting item, `null` is returned. We don't want to store `[null]` to activity